### PR TITLE
Update and test `interface_entry_point` macro

### PR DESCRIPTION
**Pull request by Kayanski**
This PR aims at : 
1. Adapt the `interface_entry_point` macro to the changed traits needed for interacting with the chain
2. Add tests for the `interface_entry_point` macro

### CODE REVIEW

File: cw-orch/tests/entry_point.rs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        .unwrap();
+    assert_eq!(response.events.len(), 1);
+}
+
+#[test]
+#[cfg(feature = "node-tests")]
+fn daemon_test() {
+    use cw_orch::prelude::networks;
+
+    let runtime = Arc::new(Runtime::new().unwrap());
```
Comment (CyberHoward) [inline cw-orch/tests/entry_point.rs:110]: this doesn't need to be Arc

File: contracts/mock_contract/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,89 @@
+use cosmwasm_std::to_binary;
+use cosmwasm_std::Deps;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::Binary;
+use cosmwasm_std::MessageInfo;
+use cosmwasm_std::StdError;
+use cosmwasm_std::Response;
+use cosmwasm_std::StdResult;
+use cosmwasm_std::DepsMut;
+use cosmwasm_std::{Env, entry_point};
```
Comment (CyberHoward) [inline contracts/mock_contract/src/lib.rs:10]: Bunch of clippy warnings.
`cargo clippy -- -D warnings`

Would you be opposed to having a pre-push hook installed?

File: cw-orch/tests/entry_point.rs (hunk 1 above)
Comment (Kayanski) [reply cw-orch/tests/entry_point.rs:110]: yes. I used what was used in other tests here

File: contracts/mock_contract/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+[package]
+name = "mock-contract"
+version = "0.11.0-alpha.5"
+description = "Mock constract for interface_entry_point macro testing"
```
[the remark this answers is not in the document]
Comment (Kayanski) [reply contracts/mock_contract/Cargo.toml:4]: sure. Do we change the existing cw20 tests to this contract also ?

**Comment by CyberHoward** [comment]
Could you either add the function macro tests in this pr as well or create a new branch/draft pr based on this branch that adds those tests and changes mentioned?

**Comment by Kayanski** [comment]
> Could you either add the function macro tests in this pr as well or create a new branch/draft pr based on this branch that adds those tests and changes mentioned?

Sure, I'm doing that right now :)

**Pull request merged**